test_manager/features/environment.py

```diff
@@ -38,18 +42,71 @@ def before_all(context):
     :return: None
     """
 
+    Scenario.continue_after_failed_step = True
+
     context.transport = {}
     context.ues = {}
-    context.dispatcher = {}
+    context.dispatcher = None
+
+    context.ue_tracker = []
 
     loggerutils.setup_logging()
     loggerutils.setup_formatted_logging(context)
 
-    test_manager = TestManager(context, "127.0.0.5", 12345)
+    context.logger.info("Setting up Test Environment...")
+
+    test_manager = TestManager(context, "127.0.0.5", 33333)
     thread = Thread(target=test_manager.listen_for_incoming_events)
     thread.start()
     context.tm = test_manager
 
+    counter = 1
+    all_languages = []
+    all_transports = []
+    all_uris = []
+    while context.config.userdata.get(f"uE{str(counter)}") is not None:
+        all_languages.append(context.config.userdata[f"uE{str(counter)}"])
+        all_transports.append(context.config.userdata[f"transport{str(counter)}"])
+        context.logger.info(all_languages)
+
+        current_uri = {
+            "authority_name": "myAuthority",
```

Comment:
Should this name be different per testAgent

Reply:
maybe? I set it now to be dynamic according to the test agent.